src/main/scala/ai/privado/exporter/ExporterUtility.scala

```diff
@@ -462,12 +443,30 @@ object ExporterUtility {
       Constants.propertyFileSkippedByDirCount -> propertyFilterCache.getFileSkippedDirCountData(ruleCache).asJson
     )
 
+    val sinkSubCategories = mutable.HashMap[String, mutable.Set[String]]()
+    ruleCache.getRule.sinks.foreach(sinkRule => {
+      if (!sinkSubCategories.contains(sinkRule.catLevelTwo))
+        sinkSubCategories.addOne(sinkRule.catLevelTwo -> mutable.Set())
+      sinkSubCategories(sinkRule.catLevelTwo).add(sinkRule.nodeType.toString)
+    })
+
+    val dataflowsOutput = mutable.LinkedHashMap[String, List[DataFlowSubCategoryModel]]()
+    sinkSubCategories.foreach(sinkSubTypeEntry => {
+      dataflowsOutput.addOne(
+        sinkSubTypeEntry._1 -> dataflowExporter
+          .getFlowByType(sinkSubTypeEntry._1, sinkSubTypeEntry._2.toSet, ruleCache, dataFlowModel, appCache)
+          .toList
+      )
+    })
+
+    val sourceExporter = new SourceExporter(cpg, ruleCache, privadoInput, repoItemTagName = repoItemTagName, appCache)
+
     // Future creates a thread and starts resolving the function call asynchronously
     val sources = Future {
       Try(sourceExporter.getSources).getOrElse(List[SourceModel]())
     }
     val processing = Future {
-      Try(sourceExporter.getProcessing).getOrElse(List[SourceProcessingModel]())
+      Try(sourceExporter.getProcessing(dataflowsOutput, dataflows)).getOrElse(List[SourceProcessingModel]())
```

Comment:
Remove Future for processing as it depends on dataflowsOutput

Reply:
Yes right.